src/renderer/features/network/CustomRpcForm/ui/EditCustomRpcModal.tsx

```diff
@@ -0,0 +1,102 @@
+import { FormEvent, useEffect } from 'react';
+import { useForm } from 'effector-forms';
+import { useUnit } from 'effector-react';
+
+import { BaseModal, Button, Input, InputHint, Alert } from '@shared/ui';
+import { useI18n } from '@app/providers';
+import { OperationTitle } from '@entities/chain';
+import { RpcCheckResult, editCustomRpcModel } from '@features/network';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const EditCustomRpcModal = ({ isOpen, onClose }: Props) => {
+  const { t } = useI18n();
+  const rpcCheckResult = useUnit(editCustomRpcModel.$rpcConnectivityResult);
+  const network = useUnit(editCustomRpcModel.$selectedNetwork);
+
+  useEffect(() => {
+    editCustomRpcModel.events.formInitiated();
+  }, []);
+
+  const {
+    fields: { name, url },
+    submit,
+    isValid,
+  } = useForm(editCustomRpcModel.$editCustomRpcForm);
+
+  const onSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    submit();
+  };
+
+  const isLoading = rpcCheckResult === RpcCheckResult.LOADING;
```

Comment:
Sure, I went for the store and no util. I know you're using a lot of utils across the code base, to me these are very repetitive since I find the enum already quite clear. lmk if you think I should add a util

Reply:
It's about business rules in UI
As long as `isLoading` comes from the store (ideally) and it's boolean we're fine, not bothering ourselves how it's being calculated (business rules)
Once business rules leak to UI we have a problem
utils function that checks enums/states/or whatever not that complicated, is just a compromise and a way to avoid introducing yet another simple boolean store
The ideal solution is utils function with business rule + store that has it's value set via that utility function (+ tests for all utility functions)

Reply:
Sounds good, I'll add a util then.